File: src/constants/revenueShares.ts
/**
 * Literium Revenue Sharing Constants
 * Centralized single source of truth for platform vs writer profit splits.
 */

export const REVENUE_SHARES = {
  // 1. Ads inside writer articles: 55% Writer / 45% Platform
  IN_ARTICLE_ADS: {
    WRITER: 0.55,
    PLATFORM: 0.45,
    WRITER_PERCENT: 55,
    PLATFORM_PERCENT: 45,
    LABEL: '55% للكاتب / 45% للمنصة'
  },

  // 2. Ads on writer personal profile page: 50% Writer / 50% Platform
  WRITER_PROFILE_ADS: {
    WRITER: 0.50,
    PLATFORM: 0.50,
    WRITER_PERCENT: 50,
    PLATFORM_PERCENT: 50,
    LABEL: '50% للكاتب / 50% للمنصة'
  },

  // 3. Locked / Premium Articles sales: 85% Writer / 15% Platform
  LOCKED_ARTICLES: {
    WRITER: 0.85,
    PLATFORM: 0.15,
    WRITER_PERCENT: 85,
    PLATFORM_PERCENT: 15,
    LABEL: '85% للكاتب / 15% للمنصة'
  },

  // 4. Platform general ads (Homepage, explore, categories): 100% Platform / 0% Writer
  PLATFORM_ADS: {
    WRITER: 0.00,
    PLATFORM: 1.00,
    WRITER_PERCENT: 0,
    PLATFORM_PERCENT: 100,
    LABEL: '100% للمنصة'
  }
} as const;

// Default AdSense writer split for blended in-article estimates (55%)
export const DEFAULT_WRITER_ADSENSE_SHARE = REVENUE_SHARES.IN_ARTICLE_ADS.WRITER;
export const DEFAULT_PLATFORM_ADSENSE_SHARE = REVENUE_SHARES.IN_ARTICLE_ADS.PLATFORM;
export const DEFAULT_WRITER_SALES_SHARE = REVENUE_SHARES.LOCKED_ARTICLES.WRITER;
export const DEFAULT_PLATFORM_SALES_SHARE = REVENUE_SHARES.LOCKED_ARTICLES.PLATFORM;

File: android-native/app/src/main/kotlin/studio/ai/literium/literium_app/util/RevenueShares.kt
package studio.ai.literium.literium_app.util

/**
 * Single source of truth for platform-vs-writer revenue splits. Ported
 * verbatim from `src/constants/revenueShares.ts`. Referenced live (not
 * hardcoded separately) from `PoliciesModal`, `LegalPages`,
 * `SmartAiGuidanceCard`, and `AdvertiserDashboard` in the web app — the
 * Kotlin app must likewise treat this object as the only place these
 * numbers live (spec §5.6/§12.2).
 */
object RevenueShares {

    data class Split(val writer: Double, val platform: Double, val writerPercent: Int, val platformPercent: Int, val label: String)

    /** Ads embedded inside a writer's own article: 55% writer / 45% platform. */
    val IN_ARTICLE_ADS = Split(0.55, 0.45, 55, 45, "55% للكاتب / 45% للمنصة")

    /** Ads shown on a writer's own profile page: 50% writer / 50% platform. */
    val WRITER_PROFILE_ADS = Split(0.50, 0.50, 50, 50, "50% للكاتب / 50% للمنصة")

    /** Locked/exclusive article direct-purchase sales: 85% writer / 15% platform. */
    val LOCKED_ARTICLES = Split(0.85, 0.15, 85, 15, "85% للكاتب / 15% للمنصة")

    /** Platform-wide ads (homepage, category pages, other platform-owned surfaces): 100% platform / 0% writer. */
    val PLATFORM_ADS = Split(0.00, 1.00, 0, 100, "100% للمنصة")

    /** Blended default AdSense writer/platform split used for in-article estimates. */
    val DEFAULT_WRITER_ADSENSE_SHARE = IN_ARTICLE_ADS.writer
    val DEFAULT_PLATFORM_ADSENSE_SHARE = IN_ARTICLE_ADS.platform
    val DEFAULT_WRITER_SALES_SHARE = LOCKED_ARTICLES.writer
    val DEFAULT_PLATFORM_SALES_SHARE = LOCKED_ARTICLES.platform

    /** Looks up the [Split] for a given [studio.ai.literium.literium_app.data.model.AdSlotId] by its beneficiary/share pair,
     *  matching `AdSlot.tsx`'s `SLOT_CONFIG` (writer slots pay 55% except the two writer-profile slots, which pay 50%). */
    fun splitForWriterSlot(writerShare: Double): Split = when (writerShare) {
        0.55 -> IN_ARTICLE_ADS
        0.50 -> WRITER_PROFILE_ADS
        else -> PLATFORM_ADS
    }
}
